File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/S. Sum of Consecutive Odd Numbers/main.dart
import 'dart:io';
import 'dart:math';

void main() {
  int t = int.parse(stdin.readLineSync()!);

  while (t-- > 0)
  {
    List<String> input = stdin.readLineSync()!.split(" ");
    final int x = int.parse(input[0]);
    final int y = int.parse(input[1]);
    print(sumAllOdds(x, y));
  }
}

int sumAllOdds(int x, int y){
  int mn = min(x, y);
  int mx = max(x, y);

  int i = ((1 & mn) == 1 ? mn + 2 : mn + 1);
  int sum = 0;

  for (; i < mx; i += 2)
  {
    sum += i;
  }

  return sum;
}

File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/S. Sum of Consecutive Odd Numbers/main.cs
// https://codeforces.com/group/MWSDmqGsZm/contest/326175/problem/S

using System;
using static System.Console;
using static System.Math;

namespace Codeforces.Assuit.Contest1
{
    class ProblemS
    {
        static void Main(string[] args)
        {
            ushort t = ushort.Parse(ReadLine());

            while (t-- > 0)
            {
                string[] input = ReadLine()!.Split(" ");
                ushort x = ushort.Parse(input[0]);
                ushort y = ushort.Parse(input[1]);
                WriteLine(SumAllOdds(x, y));
            }
        }
        static uint SumAllOdds(in ushort x, in ushort y){
            ushort mn = Min(x, y);
            ushort mx = Max(x, y);

            ushort i = (ushort)((1 & mn) == 1 ? mn + 2 : mn + 1);
            uint sum = 0;

            for (; i < mx; i += 2)
            {
                sum += i;
            }

            return sum;
        }
    }
}
